tests/quick.rs

```diff
@@ -526,6 +527,11 @@ quickcheck! {
         correct_size_hint(izip!(filt, b.clone(), c.clone())) &&
             exact_size(izip!(a, b, c))
     }
+
+    fn size_chain_macro(a: Iter<i16, Exact>, b: Iter<i16, Exact>, c: Iter<i16, Exact>) -> bool {
+        correct_size_hint(chain!(a.clone(), b.clone(), c.clone()))
+    }
```

Comment:
I hope I don't miss anything, but is `size_hint` special for `chain!`? What about other potentially specialized methods?

Reply:
It is not special. I guess this test shouldn't be there (I juste copied some of the `izip!` tests) since it tests more the properties of the std `chain`: we don't specialize any method in this crate. I don't see how the macro could be wrong while `chain` being correct.

If we were to verify all the methods that are specialized by `Chain`, that would mean writing quite many tests ([here](https://doc.rust-lang.org/nightly/src/core/iter/adapters/chain.rs.html#68) is the list).